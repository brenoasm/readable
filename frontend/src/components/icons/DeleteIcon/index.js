import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Icon from '../Icon';

const propTypes = {
  handleDelete: PropTypes.func.isRequired,
  objectToDelete: PropTypes.object,
  title: PropTypes.string
};

const defaultProps = {
  handleDelete: () => {},
  objectToDelete: {},
  title: ''
};

const StyledEditIcon = styled.span`
  cursor: pointer;
`;

const DeleteIcon = ({ handleDelete, objectToDelete, title }) => (
  <StyledEditIcon>
    <Icon
      title={title}
      icon="fas fa-trash"
      onClick={() => handleDelete(objectToDelete)}
    />
  </StyledEditIcon>
);

DeleteIcon.propTypes = propTypes;
DeleteIcon.defaultProps = defaultProps;

export default DeleteIcon;
