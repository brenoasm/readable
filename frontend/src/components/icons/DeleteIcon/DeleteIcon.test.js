/* eslint-disable */

import React from 'react';

import DeleteIcon from './';

describe('DeleteIcon', () => {
  it('matches the snapshot', () => {
    const initialProps = {
      title: 'Delete teste',
      objectToDelete: {},
      handleDelete: jest.fn()
    };

    const wrapper = shallow(<DeleteIcon {...initialProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
