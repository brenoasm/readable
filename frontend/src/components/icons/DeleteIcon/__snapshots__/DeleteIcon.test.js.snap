// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteIcon matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeleteIcon
    handleDelete={[Function]}
    objectToDelete={Object {}}
    title="Delete teste"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Icon
        icon="fas fa-trash"
        onClick={[Function]}
        title="Delete teste"
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "icon": "fas fa-trash",
        "onClick": [Function],
        "title": "Delete teste",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Icon
          icon="fas fa-trash"
          onClick={[Function]}
          title="Delete teste"
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "icon": "fas fa-trash",
          "onClick": [Function],
          "title": "Delete teste",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
